src/interp.rs

```diff
@@ -534,8 +534,7 @@ fn in_bounds<'tcx>(val: u128, ty: Ty<'tcx>) -> bool {
         TypeVariants::TyInt(int_ty) => {
             match int_ty {
                 IntTy::I32 => {
-                    let val = val as i32;
-                    val >= <i32>::min_value() && val <= <i32>::max_value()
+                    val >= <i32>::min_value() as u128 && val <= <i32>::max_value() as u128
```

Comment:
I'm surprised `<i32>::min_value() as u128` works. -2^{31} can't expressed by a `u128`.

Or am I missing something? Perhaps we should have some unit tests for the bounds check?

Reply:
I don't know how the conversion works here, but I think it might become a 32 bit pattern (i.e. it'll become something like +2^32). I agree that this feels a bit dangerous though.

Reply:
Note that Jake killed this logic in a later commit.

Reply:
Aha, OK.